src/js/lib/cachetasks.js

```diff
@@ -191,6 +191,27 @@
       }, this);
     },
     /**
+     * Update the response associated with a request.
+     *
+     * @param  {Request} request   The request to check.
+     * @param  {String} cacheName The name of the cache to use.
+     * @param  {Object} options    options
+     *                              - force: true or false
+     * @return {Promise} Resolves once operations complete.
+     */
+    update(request, cacheName, options = {force: false}) {
+      return async.task(function*() {
+        let cache = yield MemoizedCaches.open(cacheName);
+        let response = yield cache.match(request);
+        let shouldUpdate = !response || options.force || this.isStale(response);
+        if (shouldUpdate) {
+          response = yield fetch(request);
```

Comment:
Should the fetch fail, what should be the behavior?
Respond with an error and blow the cache?

In this case, I think the prudent option would be to return the cached request even if it's stale